lib/pure/times.nim

```diff
@@ -891,24 +827,33 @@ proc formatToken(info: TimeInfo, token: string, buf: var string) =
     if fyear.len != 5: fyear = repeat('0', 5-fyear.len()) & fyear
     buf.add(fyear)
   of "z":
-    let hrs = (info.timezone div 60) div 60
-    buf.add($hrs)
+    let
+      factor = if info.timezone <= 0: -1 else: 1
+      hours = (factor * info.timezone) div secondsInHour
+    if factor == 1: buf.add('-')
+    else: buf.add('+')
+    buf.add($hours)
   of "zz":
-    let hrs = (info.timezone div 60) div 60
-
-    buf.add($hrs)
-    if hrs.abs < 10:
-      var atIndex = buf.len-(($hrs).len-(if hrs < 0: 1 else: 0))
-      buf.insert("0", atIndex)
+    let
+      factor = if info.timezone <= 0: -1 else: 1
+      hours = (factor * info.timezone) div secondsInHour
+    if factor == 1: buf.add('-')
```

Comment:
Because of different semantics. In a formatted date, `+01:00` means „one hour *ahead* of UTC“, so to calculate UTC, one needs to *substract* one hour from the local time. This is what's stored in `TimeInfo.timezone`.